Coverage Report workflow chronically fails: Tests.DocExamples lacks coverlet.collector

The `Coverage Report` workflow fails on every run — including on release tags — because `Wolfgang.Etl.Transformers.Tests.DocExamples` cannot collect coverage.

```
Data collection : Unable to find a datacollector with friendly name 'XPlat Code Coverage'.
Data collection : Could not find data collector 'XPlat Code Coverage'
##[error]Process completed with exit code 1.
```

**All tests pass** — 320 unit + 12 fuzz + 1 doc-example. The job fails purely on data collection, so the red X is not a test failure.

## Cause

Only two of the seven test projects reference `coverlet.collector`:

| project | coverlet.collector |
| --- | --- |
| `Tests.Unit` | yes |
| `Tests.Integration` | yes |
| `Tests.DocExamples` | **missing** |
| `Tests.Allocation` | **missing** |
| `Tests.Concurrency` | **missing** |
| `Tests.Fuzz` | **missing** |
| `Tests.Snapshots` | **missing** |

`dotnet test --collect:"XPlat Code Coverage"` needs the collector referenced by each project it runs. `DocExamples` is the one currently reached by the coverage run, so it is the one that errors; the others are latent and will surface the same way if the run is broadened.

## Chronic, not new

| run | branch | result |
| --- | --- | --- |
| 2026-08-13 18:13 | v0.5.0 | failure |
| 2026-08-13 18:07 | main | failure |
| 2026-08-13 17:09 | v0.5.0 | failure |
| 2026-08-11 01:26 | **v0.4.0** | failure |
| 2026-08-11 00:50 | main | failure |

It was already failing when v0.4.0 shipped, so it has never blocked a release — but it means **this repo has had no published coverage figure for at least two releases**, and a genuine coverage regression would be indistinguishable from the current noise.

## Fix

Add `coverlet.collector` to `Tests.DocExamples` (and ideally the other four, so broadening the coverage run later does not re-break it):

```xml
<PackageReference Include="coverlet.collector" Version="...">
  <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
  <PrivateAssets>all</PrivateAssets>
</PackageReference>
```

Match the version already used by `Tests.Unit`.

## Acceptance criteria

- [ ] `Coverage Report` completes green on `main`.
- [ ] A coverage figure is actually published/reported, not just a green check.
- [ ] Adding a new test project does not silently reintroduce the failure (consider a shared `tests/Directory.Build.props`).

--- a/tests/Wolfgang.Etl.Transformers.Tests.Allocation/Wolfgang.Etl.Transformers.Tests.Allocation.csproj
+++ b/tests/Wolfgang.Etl.Transformers.Tests.Allocation/Wolfgang.Etl.Transformers.Tests.Allocation.csproj
@@ -25,6 +25,10 @@
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
       <PrivateAssets>all</PrivateAssets>
     </PackageReference>
+    <PackageReference Include="coverlet.collector" Version="10.0.1">
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+      <PrivateAssets>all</PrivateAssets>
+    </PackageReference>
   </ItemGroup>
 
 </Project>

--- a/tests/Wolfgang.Etl.Transformers.Tests.Concurrency/Wolfgang.Etl.Transformers.Tests.Concurrency.csproj
+++ b/tests/Wolfgang.Etl.Transformers.Tests.Concurrency/Wolfgang.Etl.Transformers.Tests.Concurrency.csproj
@@ -36,6 +36,10 @@
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
       <PrivateAssets>all</PrivateAssets>
     </PackageReference>
+    <PackageReference Include="coverlet.collector" Version="10.0.1">
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+      <PrivateAssets>all</PrivateAssets>
+    </PackageReference>
   </ItemGroup>
 
 </Project>

--- a/tests/Wolfgang.Etl.Transformers.Tests.DocExamples/Wolfgang.Etl.Transformers.Tests.DocExamples.csproj
+++ b/tests/Wolfgang.Etl.Transformers.Tests.DocExamples/Wolfgang.Etl.Transformers.Tests.DocExamples.csproj
@@ -25,6 +25,10 @@
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
       <PrivateAssets>all</PrivateAssets>
     </PackageReference>
+    <PackageReference Include="coverlet.collector" Version="10.0.1">
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+      <PrivateAssets>all</PrivateAssets>
+    </PackageReference>
   </ItemGroup>
 
 </Project>

--- a/tests/Wolfgang.Etl.Transformers.Tests.Fuzz/Wolfgang.Etl.Transformers.Tests.Fuzz.csproj
+++ b/tests/Wolfgang.Etl.Transformers.Tests.Fuzz/Wolfgang.Etl.Transformers.Tests.Fuzz.csproj
@@ -28,6 +28,10 @@
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
       <PrivateAssets>all</PrivateAssets>
     </PackageReference>
+    <PackageReference Include="coverlet.collector" Version="10.0.1">
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+      <PrivateAssets>all</PrivateAssets>
+    </PackageReference>
   </ItemGroup>
 
 </Project>

--- a/tests/Wolfgang.Etl.Transformers.Tests.Snapshots/Wolfgang.Etl.Transformers.Tests.Snapshots.csproj
+++ b/tests/Wolfgang.Etl.Transformers.Tests.Snapshots/Wolfgang.Etl.Transformers.Tests.Snapshots.csproj
@@ -29,6 +29,10 @@
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
       <PrivateAssets>all</PrivateAssets>
     </PackageReference>
+    <PackageReference Include="coverlet.collector" Version="10.0.1">
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+      <PrivateAssets>all</PrivateAssets>
+    </PackageReference>
   </ItemGroup>
 
 </Project>

